scipy/optimize/_differentialevolution.py

```diff
@@ -1343,7 +1343,17 @@ def __init__(self, f, args):
         self.args = [] if args is None else args
 
     def __call__(self, x):
-        return self.f(x, *self.args)
+        fx = self.f(np.copy(x), *self.args)
+        # Make sure the function returns a true scalar
+        if not np.isscalar(fx):
```

Comment:
With the way that the minimisation is done it should be ultimately possible to work with a vectorised `func`, which would then return a vector as well. Thus a future API change could be removing this code anyway.

It shouldn't matter if the user function returns `1, [1], np.array([[[[1]]]])` as the return from `func` is made into an array and squeezed on L901.

I think before the changes in L1347 to L1356 are committed a test should be made to see what happens when a length > 1 array is returned from `func`. I'm guessing that something will go wrong in L901, possibly falling into the except clause on L902.

It doesn't make sense to check for 'scalar' nature here, it just adds too much overhead. Can you make the test on L901 in `_calculate_population_energies`:
```
calc_energies = np.squeeze(calc_energies)
if calc_energies.size != nfevs:
    raise ValueError(
        "The user-provided objective function must return a scalar value."
    )
energies[0:nfevs] = calc_energies
```
Note that `_calculate_population_energies` is *always* called at the start of a solve cycle (L1057).

Reply:
> With the way that the minimisation is done it should be ultimately possible to work with a vectorised func, which would then return a vector as well. 

Cool! How would that work with `minimize(self.func)` (L903)? (That was the line that had me assume you always have to have a scalar-valued function.)

Reply:
Never mind. I just added `np.squeeze` in another line which also fixes the problem.